eng/scripts/Verify-PreviewDependency.ps1

```diff
@@ -4,14 +4,20 @@ param (
   [string]$PackagePath
 )
 
+."$PSScriptRoot\..\common\scripts\SemVer.ps1"
+
 $packagePaths = Get-ChildItem -Path "$PackagePath/*/package.json"
 
 foreach ($packagePath in $packagePaths) {
   $pkgContent = Get-Content $packagePath | ConvertFrom-Json
   foreach ($dependency in $pkgContent.dependencies.PSObject.properties) {
     $depVer = $dependency.Value
-    if (($depVer -match "\D0.\d.\d") -or ($depVer -match "\D\d.\d.\d-\w")) {
-      throw "Found dependency on preview/pre-release package $($dependency.Name) with version range $depVer. Please pin dependencies to preview packages."
+    if ($depVer -match "\D\d.\d.\d") {
+      $baseVer = $depVer.replace("~","").replace("^","")
```

Comment:
If we find we need this version treatment in other places I suggest adding a ParseJSDependencyVersion method to the AzureEngSemanticVersion class itself to handle these.